mountpoint-s3/tests/common/s3.rs

```diff
@@ -34,6 +34,39 @@ pub fn get_test_region() -> String {
     std::env::var("S3_REGION").expect("Set S3_REGION to run integration tests")
 }
 
+// Get a random region other than what configured in S3_REGION
+pub fn get_random_non_test_region() -> String {
+    // All AWS regions that are enabled by default, we can get this list from APIs but that's probably overkill.
+    let regions = vec![
+        "ap-northeast-1",
+        "ap-northeast-2",
+        "ap-northeast-3",
+        "ap-south-1",
+        "ap-southeast-1",
+        "ap-southeast-2",
+        "ca-central-1",
+        "eu-central-1",
+        "eu-north-1",
+        "eu-west-1",
+        "eu-west-2",
+        "eu-west-3",
+        "sa-east-1",
+        "us-east-1",
+        "us-east-2",
+        "us-west-1",
+        "us-west-2",
+    ];
+
+    let test_region = get_test_region();
+    let mut rng = rand::thread_rng();
+    regions
+        .into_iter()
+        .filter(|region| region.to_string() != test_region)
+        .choose(&mut rng)
+        .unwrap()
+        .to_string()
+}
```

Comment:
I'd rather we try to avoid randomness between tests. Instead, what if we just have something like...

```suggestion
// Get a region other than what configured in S3_REGION
pub fn get_non_test_region() -> String {
    match get_test_region() {
        "us-east-1" => String::new("us-west-2"),
        _ => String::new("us-east-1"),
    }
}
```